File: app/src/main/java/vergecurrency/vergewallet/service/model/network/layers/NetworkGateway.java
package vergecurrency.vergewallet.service.model.network.layers;


import android.os.AsyncTask;

import java.util.concurrent.ExecutionException;

import vergecurrency.vergewallet.service.model.PreferencesManager;


public class NetworkGateway extends AsyncTask<String, Integer, String> {

	@Override
	protected String doInBackground(String... strings) {
		String result;
		if (PreferencesManager.getUsingTor()) {
			//TODO : Redo tg class
			TorLayerGateway tg = TorLayerGateway.getInstance();
			result = tg.doInBackground(strings);
		} else {
			/*try {
				//TODO = fix this.
				result = new ClearnetGateway().execute(strings).get();
			} catch (ExecutionException e) {
				e.printStackTrace();
				result = "";
			} catch (InterruptedException e) {
				e.printStackTrace();
				result = "";
			}*/
			result = "";
		}


		return result;
	}
}

File: app/src/main/kotlin/vergecurrency/vergewallet/service/model/network/layers/NetworkGateway.kt
package vergecurrency.vergewallet.service.model.network.layers

import cz.msebera.android.httpclient.client.methods.HttpRequestBase
import vergecurrency.vergewallet.service.model.EncryptedPreferencesManager
import vergecurrency.vergewallet.service.model.PreferencesManager

open class NetworkGateway {


    fun doRequest(requestBase: HttpRequestBase): String {
        var result = ""
        //I want to get around tor for instance
        if (PreferencesManager.usingTor && 1 ==0 ) {
            result = TorLayerGateway(requestBase).execute().get()
        } else {
            result = ClearnetGateway(requestBase).execute().get()
        }

        return result
    }
}